app/src/test/java/com/wtmberlin/SetMainDispatcherRule.kt

```diff
@@ -0,0 +1,22 @@
+package com.wtmberlin
+
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.newSingleThreadContext
+import kotlinx.coroutines.test.resetMain
+import kotlinx.coroutines.test.setMain
+import org.junit.rules.TestWatcher
+import org.junit.runner.Description
+
+@ExperimentalCoroutinesApi
+class SetMainDispatcherRule : TestWatcher() {
+
+    override fun starting(description: Description?) {
+        Dispatchers.setMain(newSingleThreadContext("UI thread"))
```

Comment:
Don’t forget you’ll need to release this thread on test finish

Reply:
https://github.com/Kotlin/kotlinx.coroutines/tree/master/kotlinx-coroutines-test

Check the example where the surrogate thread is being closed.